en/middleware/micrortps.md

```diff
@@ -0,0 +1,344 @@
+
+```cmake
+set(config_rtps_send_topics
+   sensor_accel
+   sensor_baro
+   sensor_gyro
+   ...
+   )
+
+set(config_rtps_receive_topics
+   sensor_combined
+   telemetry_status
+   wind_estimate
+   ...
+   )
+```
+
+The client application will be generated in *build_OURPLATFORM/src/modules/micrortps_bridge/micrortps_client/* folder and the agent will be created in *src/modules/micrortps_bridge/micrortps_agent/* folder.
+
+> **Note** In the process of the agent generation we use a Fast RTPS tool called *fastrtpsgen*. If you haven't installed Fast RTPS in the default path you need to specify the directory of installation of *fastrtpsgen* setting the environment variable `FASTRTPSGEN_DIR` before executing *make*. 
+> On Linux/Mac this is done as shown below:
+>
+>  ```sh
+  export FASTRTPSGEN_DIR=/path/to/fastrtps/install/folder/bin
+  ```
+
+### Manually generate client and agent code
+  
+It is also possible to generate and install the code for the client and the agent outside the normal PX4 build process using the python script
+**Tools/generate_microRTPS_bridge.py**. 
+
+The tool's command syntax is shown below:
+
+```sh
+$ cd /path/to/PX4/Firmware
+$ python Tools/generate_microRTPS_bridge.py -h
+usage: generate_microRTPS_bridge.py [-h] [-s *.msg [*.msg ...]]
+                                    [-r *.msg [*.msg ...]] [-a] [-c]
+                                    [-t MSGDIR] [-o AGENTDIR] [-u CLIENTDIR]
+                                    [-f FASTRTPSGEN]
+
+optional arguments:
+  -h, --help            show this help message and exit
+  -s *.msg [*.msg ...], --send *.msg [*.msg ...]
+                        Topics to be sended
+  -r *.msg [*.msg ...], --receive *.msg [*.msg ...]
+                        Topics to be received
+  -a, --agent           Flag for generate the agent, by default is true if -c
+                        is not specified
+  -c, --client          Flag for generate the client, by default is true if -a
+                        is not specified
+  -t MSGDIR, --topic-msg-dir MSGDIR
+                        Topics message dir, by default msg/
+  -o AGENTDIR, --agent-outdir AGENTDIR
+                        Agent output dir, by default
+                        src/modules/micrortps_bridge/micrortps_agent
+  -u CLIENTDIR, --client-outdir CLIENTDIR
+                        Client output dir, by default
+                        src/modules/micrortps_bridge/micrortps_client
+  -f FASTRTPSGEN, --fastrtpsgen-dir FASTRTPSGEN
+                        fastrtpsgen installation dir, by default /bin
+  --no-delete           Do not delete dir tree output dir(s)
+```
+
+- The argument `--send/-s` means that the application from PX4 side will send these messages, and the argument `--receive/-r` specifies which messages is going to be received.
```

Comment:
"messages is going" should be "messages are going"

So you can declare multiple messages for -s and -r? Can you call this multiple times to generate the new code?

I also assume that if you do this you don't need the .cmake changes?